### PR TITLE
docs: modernize patterns in subset of samples

### PR DESCRIPTION
**Pull request by JustinBeckwith**
This doesn't capture everything, but it does get the quickstart and a few other samples.

### CODE REVIEW

File: samples/quickstart.js [hunk 1]
```diff
@@ -14,42 +14,51 @@
 
 'use strict';
 
-// [START speech_quickstart]
 async function main() {
```
Comment (sofisl) [inline samples/quickstart.js:17]: Should this just be function main()? I think we are still bubbling up the correct error on lines 59 - 62.

Comment (JustinBeckwith) [reply samples/quickstart.js]: Yeah good call.  I switched this up in a few places, and hopefully cleaned up the failing tests.

File: samples/system-test/MicrophoneStream.test.js [hunk 2]
```diff
@@ -1,35 +0,0 @@
...
-const cp = require('child_process');
-
-const execSync = cmd => cp.execSync(cmd, {encoding: 'utf-8'});
-
-const cmd = 'node MicrophoneStream.js';
-const cwd = path.join(__dirname, '..');
-
-describe('MicrophoneStream', () => {
-  it('should load and display Yaaaarghs(!) correctly', async () => {
-    const stdout = execSync(`${cmd} --help`, {cwd});
```
Comment (JustinBeckwith) [inline samples/system-test/MicrophoneStream.test.js:29]: For reviewers - the only thing this test did was check that the help text from yargs worked.  The actual sample requires `sox`, which is a native OS install, and a microphone.  In other words - very, very hard to test for real.

**Review by sofisl — APPROVED** [review]

**Comment by sofisl** [comment]
@JustinBeckwith, thank you for doing this!

**Pull request merged**